### PR TITLE
Taking data from db not frontend at OTP verification

### PR DESCRIPTION
**Pull request by Shikhar15606**
@Abhishek357 PTAL. Thanks!

### CODE REVIEW

File: api/routes/api.js [hunk 1]
```diff
@@ -61,14 +54,12 @@ router.get('/regVoter/sendOTP', fetchVoter, async (req, res, next) => {
...
     res.status(200).json({
       msg: 'Wrong phone number :(',
-      phonenumber: req.phone,
-      data,
     });
   }
 });
 
 // Verify OTP
-router.post('/regVoter/verifyOTP', async (req, res, next) => {
```
Comment (Abhishek357) [inline api/routes/api.js:71]: any particular reason for removing async?

File: api/routes/api.js [hunk 2]
```diff
@@ -61,14 +54,12 @@ router.get('/regVoter/sendOTP', fetchVoter, async (req, res, next) => {
   } else {
     res.status(200).json({
       msg: 'Wrong phone number :(',
-      phonenumber: req.phone,
```
Comment (Abhishek357) [inline api/routes/api.js:64]: we might use this data to show appropriate message in frontend

File: api/routes/api.js [hunk 3]
```diff
@@ -29,17 +29,11 @@ const init = async () => {
   );
 };
 
-router.get('/', (req, res, next) => res.json({ msg: 'Application Running' }));
+router.get('/', (req, res) => res.json({ msg: 'Application Running' }));
 
 // Register Voter in Blockchain
 // Send OTP to Voter
-router.get('/regVoter/sendOTP', fetchVoter, async (req, res, next) => {
-  if (req.hasRegistered === true) {
```
Comment (Abhishek357) [inline api/routes/api.js:37]: why are we removing this check?

File: api/middlewares/auth.js [hunk 4]
```diff
@@ -31,22 +31,37 @@ exports.checkCredentials = async (req, res, next) => {
...
+  Voters.findOne(query).exec(async (error, voterData) => {
     if (error) {
       // some error occured
       return res.status(400).json({ error });
     }
     if (voterData) {
-      // VoterID is correct checking for password
-      req.phone = voterData.phone;
-      req.district = voterData.pinCode;
-      req.hasRegistered = voterData.hasRegistered;
```
Comment (Abhishek357) [inline api/middlewares/auth.js:46]: similar doubt

**Review by Abhishek357 — COMMENTED** [review]
Overall looks good, just have some questions and nits
- A brief description of what this PR does would be great.
- We can remove the unnecessary console statements.

File: api/routes/api.js (hunk 1 above)
Comment (Shikhar15606) [reply api/routes/api.js:71]: As we are not awaiting any promises using the await keyword. We are using the .then syntax so I removed it as it served no purpose.

File: api/routes/api.js (hunk 2 above)
Comment (Shikhar15606) [reply api/routes/api.js:64]: We were not showing any message as of now. If we will show some message in future then we may add it.

File: api/routes/api.js (hunk 3 above)
Comment (Shikhar15606) [reply api/routes/api.js:37]: If you look carefully then you will find that I have added that check in the middleware itself. I tried to use the same middleware for sending and verifying OTP.

File: api/middlewares/auth.js (hunk 4 above)
Comment (Shikhar15606) [reply api/middlewares/auth.js:46]: They are there but I have moved them in an if check. PTAL

**Comment by Shikhar15606** [comment]
> Overall looks good, just have some questions and nits
> 
> * A brief description of what this PR does would be great.
> * We can remove the unnecessary console statements.

I think It would be better to keep a few console logs, they will help the first time contributors understand what is going on.

**Pull request merged**